Makefile

```diff
@@ -40,34 +51,67 @@ ALL_SRCS = \
 	test_SimulationIO.cpp
 PYTHON_EXE = _H5.so _RegionCalculus.so _SimulationIO.so
 ALL_EXE = \
-	$(PYTHON_EXE) \
-	benchmark convert-carpet-output list example \
+	sio-benchmark sio-convert-carpet-output sio-list sio-example \
 	test_RegionCalculus test_SimulationIO
 
-HDF5_DIR = /opt/local
-HDF5_CPPFLAGS = -I$(HDF5_DIR)/include
-HDF5_CXXFLAGS =
-HDF5_LDFLAGS = -L$(HDF5_DIR)/lib -Wl,-rpath,$(HDF5_DIR)/lib
-HDF5_LIBS = -lhdf5_cpp -lhdf5
-
-MPI_DIR = /opt/local
-MPI_CPPFLAGS = -I$(MPI_DIR)/include/openmpi-gcc5
-MPI_CXXFLAGS =
-MPI_LDFLAGS = -L$(MPI_DIR)/lib/openmpi-gcc5 -Wl,-rpath,$(MPI_DIR)/lib/openmpi-gcc5
-MPI_LIBS = -lmpi_cxx -lmpi
-
-PYTHON_DIR = /opt/local/Library/Frameworks/Python.framework/Versions/2.7
-PYTHON_CPPFLAGS = -I$(PYTHON_DIR)/include/python2.7
-PYTHON_CXXFLAGS =
-PYTHON_LDFLAGS = -L$(PYTHON_DIR)/lib -Wl,-rpath,$(PYTHON_DIR)/lib
-PYTHON_LIBS = -lpython2.7
-
-GTEST_DIR = googletest-release-1.7.0
+HDF5_DIR ?= $(DEFAULT_HDF5_DIR)
```

Comment:
Yes, it has to be `make VAR=value`.

Environment variables have the problem that you never really know which ones are set, and which ones the Makefile examines, so you easily get into trouble. Thus many people prefer passing settings explicitly.

Compare to calling a Python function with arguments. This can be tedious, if there are many, a single function call might require multiple lines, etc. It's then much easier to just set a few global variables that the function can examine, in particular if you need to call the function multiple times. The disadvantage is the usual disadvantage of global variables.

If things get too complicated for a single `make` call, then the usual remedy is to put the settings into a file that is included by `make`. This way, the settings remain there, permanent, even if you log out and lose your environment variable settings.